docs/book/validator.md

```diff
@@ -33,3 +34,65 @@ $validator->isValid('myIdentity', [
      'myCredentialContext' => 'myCredential',
 ]);
 ```
+
+## Validation messages
+
+Authentication validator defines five failure message types. Identifiers
+for them are available as constants for convenience.
+Common authentication failure codes, defined as constants in
+`Zend\Authentication\Result`, are mapped to validation messages
+using map in `CODE_MAP` constant. Other authentication codes default to
+`general` message type.
+
+```php
+namespace Zend\Authentication\Validator;
+
+use Zend\Authentication\Result;
+
+class Authentication
+{
+    const IDENTITY_NOT_FOUND = 'identityNotFound';
+    const IDENTITY_AMBIGUOUS = 'identityAmbiguous';
+    const CREDENTIAL_INVALID = 'credentialInvalid';
+    const UNCATEGORIZED      = 'uncategorized';
+    const GENERAL            = 'general';
+
+    const CODE_MAP = [
+        Result::FAILURE_IDENTITY_NOT_FOUND => self::IDENTITY_NOT_FOUND,
+        Result::FAILURE_CREDENTIAL_INVALID => self::CREDENTIAL_INVALID,
+        Result::FAILURE_IDENTITY_AMBIGUOUS => self::IDENTITY_AMBIGUOUS,
+        Result::FAILURE_UNCATEGORIZED      => self::UNCATEGORIZED,
+    ];
+}
+```
+
+Authentication validator extends `Zend\Validator\AbstractValidator` and provides
+common for all framework validators way to access, change or translate message templates.  
```

Comment:
```suggestion
a common way for all framework validators to access, change or translate message templates.  
```

Reply:
With this change I read it as "authentication validator provides a common way to access, ....".

May be it should be changed to "The authentication validator extends `Zend\Validator\AbstractValidator`, providing a common way for all ..." or even rephrased?

Reply:
That latter sounds perfect.

Reply:
```suggestion
a way common for all framework validators to access, change or translate message templates.  
```